controllers/topology/internal/extensions/patches/api/interface.go

```diff
@@ -78,7 +78,7 @@ type TemplateRef struct {
 	MachineDeploymentRef MachineDeploymentRef
 }
 
-func (t *TemplateRef) String() string {
+func (t TemplateRef) String() string {
```

Comment:
It seems like the `String` func should be implemented with a value receiver. That way it works independent of if the TemplateRef is a pointer or not (I tested it with `fmt.Sprintf("%s %s", &api.TemplateRef{..}, api.TemplateRef{..})`

This leads to the question if we should fix it in other cases too:
* func (n *Node) String() string {
    *  exported type in test/infrastructure/docker/docker/types
* func (p *ProviderID) String() string {
    *  exported type in controllers/noderefutil
* func (t *healthCheckTarget) string() string {
    * unexported type in controllers/
* func (n *NetworkRanges) String() string {
    *  exported type in api/v1beta1

Reply:
Let's open an issue for this

Reply:
Done: https://github.com/kubernetes-sigs/cluster-api/issues/5755